File: ColyseusTechDemo-MMO/Assets/Scripts/Models/Vectors.cs
/// <summary>
/// Object to represent <see cref="UnityEngine.Vector3"/> across messages sent to the room on the server/>
/// </summary>
public class Vector3Obj
{
    public float x;
    public float y;
    public float z;

    public Vector3Obj(float x = 0, float y = 0, float z = 0)
    {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}

/// <summary>
/// Object to represent <see cref="UnityEngine.Vector2"/> across messages sent to the room on the server/>
/// </summary>
public class Vector2Obj
{
    public float x;
    public float y;

    public Vector2Obj(float x = 0, float y = 0)
    {
        this.x = x;
        this.y = y;
    }
}

File: Server/src/helpers/Vectors.ts

export class Vector{

}

export class Vector3 extends Vector {
    x: number;
    y: number;
    z: number;

    constructor(x: number = 0, y: number = 0, z: number = 0) {
        super();

        this.x = x;
        this.y = y;
        this.z = z;
    }
}

export class Vector2 extends Vector {
    x: number;
    y: number;

    constructor(x: number = 0, y: number = 0){
        super();

        this.x = x;
        this.y = y;
    }
}